scripts/linters/pylint_extensions_test.py

```diff
@@ -3272,3 +3272,71 @@ def some_random_function(param1, param2):
 
         with self.checker_test_object.assertNoMessages():
             self.checker_test_object.checker.visit_functiondef(def_node)
+
+
+class MetaclassCheckerTests(unittest.TestCase):
+
+    def test_wrong_metaclass_usage_raises_error(self):
+        checker_test_object = testutils.CheckerTestCase()
+        checker_test_object.CHECKER_CLASS = (
+            pylint_extensions.MetaclassChecker)
+        checker_test_object.setup_method()
+
+        metaclass_node = astroid.extract_node(
+            """
+            class FakeClass(python_utils.OBJECT):
+                def __init__(self, fake_arg):
+                    self.fake_arg = fake_arg
+                def fake_method(self, name):
+                    yield (name, name)
+            class MyObject: #@
+                __metaclass__ = FakeClass
+                def __init__(self, fake_arg):
+                    self.fake_arg = fake_arg
+            """)
+
+        with checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='replace-disallowed-metaclass-usage',
```

Comment:
Do you see a way to test the message string?

Reply:
Changed the LOGIC